# HG changeset patch
# User Kris Thielemans
# Date 1221977667 -7200
#      Sun Sep 21 08:14:27 2008 +0200
# Branch release-3-0-x
# Node ID c6c678875825fd2c7e8c566be069d8bda9a3196f
# Parent  28b0618cf67c64eb8195755c25ed28fdc269f65c
For for extrapval in interpn

diff --git a/doc/interpreter/contributors.in b/doc/interpreter/contributors.in
--- a/doc/interpreter/contributors.in
+++ b/doc/interpreter/contributors.in
@@ -183,16 +183,17 @@ Christoph Spiel
 Richard Stallman
 Russell Standish
 Doug Stewart
 Thomas Stuart
 John Swensen
 Ariel Tankus
 Georg Thimm
 Duncan Temple Lang
+Kris Thielemans
 Olaf Till
 Thomas Treichl
 Utkarsh Upadhyay
 Stefan van der Walt
 Peter Van Wieren
 James R. Van Zandt
 Gregory Vanuxem
 Ivana Varekova
diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-09-18  Kris Thielemans  <kris.thielemans@csc.mrc.ac.uk>
+
+	* general/interpn.m: Fixe for extrapval and documentation of 
+	extrapval.
+
 2008-09-19  David Bateman  <dbateman@free.fr>
 
 	* general/cell2mat.m: Specialize 2D case for speed.
 
 2008-09-08  Tatsuro MATSUOKA  <tmacchant@yahoo.co.jp>
 
 	* plot/plot.m: Doc fix.
 
diff --git a/scripts/general/interpn.m b/scripts/general/interpn.m
--- a/scripts/general/interpn.m
+++ b/scripts/general/interpn.m
@@ -50,19 +50,19 @@
 ## Cubic interpolation from four nearest neighbours (not implemented yet).
 ## @item 'spline'
 ## Cubic spline interpolation--smooth first and second derivatives
 ## throughout the curve.
 ## @end table
 ##
 ## The default method is 'linear'.
 ##
-## If @var{extrap} is the string 'extrap', then extrapolate values beyond
-## the endpoints.  If @var{extrap} is a number, replace values beyond the
-## endpoints with that number.  If @var{extrap} is missing, assume NA.
+## If @var{extrapval} is the scalar value, use it to replace the values
+## beyond the endpoints with that number. If @var{extrapval} is missing,
+## assume NA.
 ## @seealso{interp1, interp2, spline, ndgrid}
 ## @end deftypefn
 
 function vi = interpn (varargin)
 
   method = "linear";
   extrapval = NA;
   nargs = nargin;
@@ -74,16 +74,17 @@ function vi = interpn (varargin)
   if (ischar (varargin{end}))
     method = varargin{end};
     nargs = nargs - 1;
   elseif (ischar (varargin{end - 1}))
     if (! isnumeric (varargin{end}) || ! isscalar (varargin{end}))
       error ("extrapal is expected to be a numeric scalar");
     endif
     method = varargin{end - 1};
+    extrapval = varargin{end};
     nargs = nargs - 2;
   endif
 
   if (nargs < 3)
     v = varargin{1};
     m = 1;
     if (nargs == 2)
       m = varargin{2};
